packages/vats/src/nameHub.js

```diff
@@ -60,40 +60,73 @@ export const makeNameHubKit = () => {
     },
   });
 
+  /** @type {LegacyMap<string, NameRecord>} */
+  // Legacy because a promiseKit is not a passable
+  const keyToAdminRecord = makeLegacyMap('nameKey');
+
   /** @type {NameAdmin} */
   const nameAdmin = Far('nameAdmin', {
     reserve(key) {
       assert.typeof(key, 'string');
-      if (!keyToRecord.has(key)) {
-        keyToRecord.init(key, makePromiseKit());
+      for (const map of [keyToAdminRecord, keyToRecord]) {
+        if (!map.has(key)) {
+          map.init(key, makePromiseKit());
+        }
       }
     },
-    update(key, newValue) {
+    update(key, newValue, adminValue) {
       assert.typeof(key, 'string');
-      const record = harden({ value: newValue });
-      if (keyToRecord.has(key)) {
-        const old = keyToRecord.get(key);
-        if (old.resolve) {
-          old.resolve(newValue);
+      /** @type {[LegacyMap<string, NameRecord>, unknown][]} */
+      const valueMapEntries = [
+        [keyToAdminRecord, adminValue], // The optional admin goes in the admin record.
+        [keyToRecord, newValue], // The value goes in the normal record.
+      ];
+      for (const [map, value] of valueMapEntries) {
+        const record = harden({ value });
+        if (map.has(key)) {
+          const old = map.get(key);
+          if (old.resolve) {
+            old.resolve(value);
+          }
+          map.set(key, record);
+        } else {
+          map.init(key, record);
         }
-        keyToRecord.set(key, record);
-      } else {
-        keyToRecord.init(key, record);
       }
     },
+    async lookupAdmin(...path) {
```

Comment:
A separate `lookupAdmin` method? hm. This structure reminds me of the [Emily](https://www.hpl.hp.com/techreports/2006/HPL-2006-116.pdf) file API.

Ah... actually, it had `subEdFiles` and `subRdFiles` just like we have `lookup` and `lookupAdmin` here. Fair enough.


```ocaml
type readable = {
isDir : unit -> bool;
exists : unit -> bool;
subRdFiles : unit -> readable list;
subRdFile : string -> readable;
inChannel : unit -> in_channel;
getBytes : unit -> string;
fullPath : unit -> string;
}
type editable = {
ro : readable;
subEdFiles : unit -> editable list;
subEdFile : string -> editable;
outChannel : unit -> out_channel;
setBytes : string -> unit;
mkDir : unit -> unit;
createNewFile : unit -> unit;
delete : unit -> unit;
}
```